### PR TITLE
core/exec.c: Implement MidOp

### LINKED ISSUE #92: Missing opcode: Mid
Implement the missing opcode and add a unit test.

### PR DESCRIPTION
**Pull request by Sebastian-byte**
Fixes #92

### CODE REVIEW

File: core/exec.c [hunk 1]
```diff
@@ -963,7 +963,59 @@ static lai_api_error_t lai_exec_reduce_op(int opcode, lai_state_t *state, struct
...
+        }
 
+        if (n >= strl) {
+            sz = 0;
+        } else if ((n + sz) > strl) {
+            sz = strl - n;
+        }
+
+        switch (object.type) {
+            case LAI_STRING: {
```
Comment (thomtl) [inline core/exec.c:990]: Does MidOp return a buffer for a string?

File: core/exec.c [hunk 2]
```diff
@@ -963,7 +963,59 @@ static lai_api_error_t lai_exec_reduce_op(int opcode, lai_state_t *state, struct
         lai_operand_emplace(state, &operands[2], &result);
         break;
     }
+    case MID_OP: {
+        lai_variable_t object = {0};
```
Comment (thomtl) [inline core/exec.c:967]: lai_variable_t's should be created with `LAI_CLEANUP_VAR lai_variable_t obj = LAI_VAR_INITIALIZER;`

File: core/exec.c [hunk 3]
```diff
@@ -963,7 +963,59 @@ static lai_api_error_t lai_exec_reduce_op(int opcode, lai_state_t *state, struct
...
+        size_t sz = length.integer;
+        if (object.type == LAI_STRING) {
+            strl = lai_exec_string_length(&object);
+        } else if (object.type == LAI_BUFFER) {
+            strl = lai_exec_buffer_size(&object);
+        }
 
+        if (n >= strl) {
+            sz = 0;
+        } else if ((n + sz) > strl) {
```
Comment (thomtl) [inline core/exec.c:985]: `Otherwise, if Index + Length is greater than or equal to the length of the buffer` Shouldn't it be >= here?

**Review by thomtl — CHANGES_REQUESTED** [review]
Few small notes, looks good otherwise I think. How did you test this? Did you write a lai_tools test? If so it would be great if you could also PR that

File: core/exec.c (hunk 3 above)
Comment (Sebastian-byte) [reply core/exec.c:985]: I'm unsure, [ACPICA seems to use greater than](https://github.com/acpica/acpica/blob/eb423b7d5440472d0d2115cb81b52b1b7c56d95a/source/components/executer/exoparg3.c#L328).

File: core/exec.c (hunk 2 above)
Comment (Sebastian-byte) [reply core/exec.c:967]: I probably got confused by how is created in some other places of LAI 😅

**Comment by Sebastian-byte** [comment]
I will write tests. Also please squash before merging, did a lot of commits 😅

File: core/exec.c (hunk 1 above)
Comment (Sebastian-byte) [reply core/exec.c:990]: It does return a string for a string, fixed it.

Comment (Sebastian-byte) [reply core/exec.c:990]: It does return a string for a string, fixed it.

**Comment by Sebastian-byte** [comment]
I will write tests.

**Review by thomtl — APPROVED** [review]

**Pull request merged**